nova_vm/src/small_string.rs

```diff
@@ -0,0 +1,43 @@
+#[derive(Clone, Copy)]
+pub struct SmallString {
+    data: [u8; 7],
+}
+
+impl std::fmt::Debug for SmallString {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "\"{}\"", self.as_str())
+    }
+}
+
+impl SmallString {
+    pub fn len(&self) -> usize {
+        self.data.iter().position(|byte| *byte == 0).unwrap_or(7)
+    }
+
+    pub fn as_bytes(&self) -> &[u8] {
+        &self.data[0..self.len()]
+    }
+
+    pub fn as_str(&self) -> &str {
+        // SAFETY: Guaranteed to be valid UTF-8.
+        unsafe { std::str::from_utf8_unchecked(self.as_bytes()) }
+    }
+
+    pub(crate) fn from_str_unchecked(value: &str) -> Self {
+        assert!(value.len() < 8);
+        let mut data: [u8; 7] = [0, 0, 0, 0, 0, 0, 0];
+        data.copy_from_slice(value.as_bytes());
+        Self { data }
+    }
+}
+
+impl TryFrom<&str> for SmallString {
+    type Error = ();
+    fn try_from(value: &str) -> Result<Self, Self::Error> {
+        if value.len() < 8 {
```

Comment:
**issue:** The string cannot end in a null character. We cannot tell apart `"foo\0"` and `"foo"` since we think the string ends at the last non-intermediate null byte.

Reply:
Thank you

Reply:
Yeah my implementation is broken. I had worked on it before you pushed and (wrongly) didn't look into it. Will copy your code.